Vagrantfile

```diff
@@ -0,0 +1,35 @@
+Vagrant.configure("2") do |config|
+  config.vm.box = "hashicorp/bionic64"
+  config.vm.provision "shell", privileged: false, inline: <<-SHELL
+    sudo apt-get update
+
+    #install pyenv prereqs
+    sudo apt-get install -y build-essential libssl-dev zlib1g-dev libbz2-dev \
+    libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev \
+    xz-utils tk-dev libffi-dev liblzma-dev python-openssl git
+
+    #install pyenv
+    git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+    echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.bash_profile
+    echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.bash_profile
+    echo -e 'if command -v pyenv 1>/dev/null 2>&1; then\n  eval "$(pyenv init -)"\nfi' >> ~/.bash_profile
+    pyenv install 2.7.8
+    pyenv install 3.8.5
+    pyenv global 3.8.5
```

Comment:
The first time this script runs, when it reaches these lines it will fail because it pyenv isn't a recognised command (i.e. not on the PATH)

You need to run `source ~/.profile` (or `source ~/.bash_profile`) between updating the file and trying to use pyenv.

Retrying with `vagrant up --provision` would work but we want `vagrant up` to work first time. You can test it by destroying your VM with `vagrant destroy` and then creating it again from scratch.

Reply:
oh nice I was wondering why the same thing worked the second time, ty